### PR TITLE
Added the --clean results option to the toolset

### PR DESCRIPTION
**Pull request by nbrady-techempower**
@msmith-techempower What do you think of this? I set it up so I could easily add `--clean installs` and `--clean all` as well, but right now `toolset/run-tests.py --clean results` will remove the results directory

### CODE REVIEW

File: toolset/benchmark/benchmarker.py [hunk 1]
```diff
@@ -963,6 +964,8 @@ def __init__(self, args):
 
     # setup results and latest_results directories 
     self.result_directory = os.path.join("results", self.name)
+    if args['clean'] == 'results':
+        shutil.rmtree(self.result_directory)
```
Comment (msmith-techempower) [inline toolset/benchmark/benchmarker.py]: I'm fine with adding `--clean`.

Broader question: does it make sense for `--clean` to also nuke the installs directory?

**Comment by nbrady-techempower** [comment]
@msmith-techempower i think in most cases, while you're testing the framework, you just want to remove the results dir because reinstalling can add a ton of time, that's why i suggested the 3 different options for `--clean [results, installs, all]`

**Comment by msmith-techempower** [comment]
I see the comment now.

After thinking about it, I think we should just add two flags: `--clean` and `--clean-all`. Your current implementation of `--clean` is great, and `--clean-all` would also include the installs directory.

**Comment by nbrady-techempower** [comment]
@msmith-techempower is there some travis config that needs to be done to recognize additional arguments? this works locally but bombs on travis

**Comment by nbrady-techempower** [comment]
@msmith-techempower this should be good to go now. I added that quick fix for vagrant users, and was sure to do a check if the directory existed first, just in case it caused any errors

**Comment by msmith-techempower** [comment]
Looks good; LGTM!

**Pull request merged**